### PR TITLE
Implement OAI-PMH for MARC21

### LINKED ISSUE #3495: Implement MARC21 format for OAI-PMH for Cicognara Sync

### PR DESCRIPTION
**Pull request by tpendragon**
Closes #3495.

### CODE REVIEW

File: app/models/oai_provider.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+class OaiProvider < OAI::Provider::Base
+  repository_name "OaiProvider"
```
Comment (hackmastera) [inline app/models/oai_provider.rb]: This looks like maybe a default value that we should change?

File: app/models/oai_provider/valkyrie_provider_model.rb [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      )
+    end
+
+    def next_set(token_string)
+      token = OAI::Provider::ResumptionToken.parse(token_string)
+      options = { last: 0 }.merge(token.to_conditions_hash)
+      find_all(options)
+    end
+
+    def change_set_persister
```
Comment (hackmastera) [inline app/models/oai_provider/valkyrie_provider_model.rb]: It's sort of confusing for this to be in here when it looks like you're only using the query service

**Comment by tpendragon** [comment]
It threw an error in staging: https://app.honeybadger.io/projects/53391/faults/57564738

**Comment by tpendragon** [comment]
This should be ready for review again.

File: Gemfile [hunk 3]
```diff
@@ -131,3 +131,5 @@ end
 
 gem "aws-sdk-s3"
 gem "extra_extra"
+
+gem "oai", "~> 1.0.0.beta1"
```
[the remark this answers is not in the document]
Comment (hackmastera) [reply Gemfile]: Also, just noticed the choice of `oai` not `blacklight_oai_provider`. Did you evaluate both? Just curious what you found out.

**Comment by tpendragon** [comment]
@hackmastera Good catches! Thanks, updated the PR.

File: Gemfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (tpendragon) [reply Gemfile]: When I started `oai` was just more recently updated and it seemed like it'd work from Postgres, meaning if solr went down it'd still be fine.

File: app/queries/paged_all_query.rb [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  delegate :connection, to: :adapter
+  delegate :orm_class, to: :resource_factory
+  def initialize(query_service:)
+    @query_service = query_service
+  end
+
+  def paged_all(limit: 10, offset: 0, from: nil, until_time: nil, only_models: [], collection_slug: nil, marc_only: false)
+    connection.transaction(savepoint: true) do
+      collections = collections(collection_slug)
+      return [] if collections.empty?
```
[the remark this answers is not in the document]
Comment (tpendragon) [reply app/queries/paged_all_query.rb]: Good catch, fixed.

**Pull request merged**